woocommerce/wc-admin
# Move WC Pay welcome page from wc-calypso-bridge

Fixes #8062 #8065 #8069 #8064 #8063 #8066 #8068

This PR moves WC Pay welcome page from https://github.com/Automattic/wc-calypso-bridge to WooCommece Admin repository.

This PR is heavy as we basically copied and pasted a directory from the bridge repository to WC Admin. The functionalities remain the same.

It adds a new menu with a red badge (1), which renders a promotional WC Pay page.

### What's changed?

1. Renamed [wc_calypso_bridge_payments_view_welcome_timestamp](https://github.com/Automattic/wc-calypso-bridge/blob/master/includes/payments/class-wc-payments-controller.php#L118) option to [wc_pay_welcome_page_viewed_timestamp](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-8a5274bc250a0e698f2589153a55b39c9bd845750ecbb4bd00d948cb37515d6dR236)

2. Renamed [wc_calypso_bridge_payments_dismissed](https://github.com/Automattic/wc-calypso-bridge/blob/master/includes/class-wc-calypso-bridge-payments.php#L105) option to [wc_pay_welcome_page_dismissed](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-d8802a4c297c546390c2dab2592c973e597640c57cc866e6ca8acf324d6dec7aR49)

3. Added [wc-pay-welcome-page](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-80aaf6db9940786bf93912c10dbeebda8dea619e4751e0d21ee09384e0a7b9f0R22) flag. 

### Things to do

1. This PR doesn't activate WC Pay promotion when the install button is clicked. I've [commented](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-8a5274bc250a0e698f2589153a55b39c9bd845750ecbb4bd00d948cb37515d6dR128) it out for now.  I think this can be done in this [issue](https://github.com/woocommerce/woocommerce-admin/issues/8067)

2. ~~I encountered an issue with webpack file watcher when [line A](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-e4275b24ae856abc29f82773797c3e4340370623daa73648319315168423d4deR196) and [line B](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-e4275b24ae856abc29f82773797c3e4340370623daa73648319315168423d4deR217) were enabled. It took about 5~10 secs to compile my changes. I've commented them out for now.~~ Done in d74f521

### Screenshots
![Screen Shot 2021-12-23 at 8 21 36 PM](https://user-images.githubusercontent.com/4723145/147315890-c0467bfc-1782-413d-998b-7bad64d69f26.jpg)
![Screen Shot 2021-12-23 at 8 21 29 PM](https://user-images.githubusercontent.com/4723145/147315900-1f265d3e-ac79-4140-94f4-7d09261e7402.jpg)

### Detailed test instructions:

1. Clone this repository.
2. Run `npm run build:feature-config`. This copies `wc-pay-welcome-page` flag to PHP.
3. Make sure WC Payments plugin is not active
4. Finish OBW and choose one of the [supported countries](https://github.com/woocommerce/woocommerce-admin/pull/8083/files#diff-7145ddc2c7845c88947923070a33b524239234f07fde79cfde7be709492464bcR31)
4. Go to the admin area and confirm a new menu with the label `Payments` and a red badge.
5. Click the menu.


Please refer to the PR linked in each issue for the additional test instructions.
